lib/vanagon/component/source/git.rb

```diff
@@ -5,20 +5,30 @@ class Component
     class Source
       class Git
         include Vanagon::Utilities
-        attr_accessor :url, :ref, :workdir, :version, :cleanup
+        attr_accessor :url, :ref, :workdir, :version, :cleanup, :clone_args
 
         # Constructor for the Git source type
         #
         # @param url [String] url of git repo to use as source
         # @param ref [String] ref to checkout from git repo
         # @param workdir [String] working directory to clone into
-        def initialize(url, ref, workdir)
+        def initialize(url, ref, workdir, clone_args = nil)
           unless ref
             fail "ref parameter is required for the git source"
           end
           @url = url
           @ref = ref
           @workdir = workdir
+          @clone_args = clone_args
+        end
+
+
+        def get_clone_args()
```

Comment:
There's already an accessor defined for `clone_args`, so a separate accessor isn't needed.When `get_clone_args` is called in `fetch`, it'll cast to an empty string if it's `nil`.